src/utils/clientHelpers.ts

```diff
@@ -1,18 +1,53 @@
-import { Executable, InitializeResult } from 'vscode-languageclient/node';
+import * as net from 'net';
+import * as vscode from 'vscode';
+import { Executable, InitializeResult, ServerOptions, StreamInfo } from 'vscode-languageclient/node';
 import { config } from './vscode';
 import { ServerPath } from './serverPath';
 
-export async function getServerExecutable(lsPath: ServerPath): Promise<Executable> {
+export async function getServerOptions(lsPath: ServerPath): Promise<ServerOptions> {
+  let serverOptions: ServerOptions;
+
+  const port: number | undefined = config('terraform').get('experimentalFeatures.languageServer.tcp.port');
+  if (port) {
+    const inspect = vscode.workspace.getConfiguration('terraform').inspect('languageServer.path');
+    if (inspect !== undefined) {
+      ///
+      if (inspect.globalValue || inspect.workspaceFolderValue || inspect.workspaceValue) {
+        vscode.window.showWarningMessage(
+          'You cannot use experimentalFeatures.languageServer.tcp.port with terraform.languageServer.path. Ignoring terraform.languageServer.path and proceeding to connect via TCP',
+        );
+      }
+    }
+
```

Comment:
```suggestion
    outputChannel.appendLine(`Connecting to language server via TCP at localhost:%d`, port);
```
I guess we'd need to get the `outputChannel` somewhere down here, but I'd say it's worth doing it as this particular line proved useful for debugging in the past - i.e. knowing how the server was launched, with what arguments etc.